src/Campaigns/Routes/CampaignOverviewStatistics.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace Give\Campaigns\Routes;
+
+use DateInterval;
+use DatePeriod;
+use DateTimeImmutable;
+use Give\API\RestRoute;
+use Give\Campaigns\CampaignDonationQuery;
+use Give\Campaigns\Models\Campaign;
+use Give\Framework\Support\Facades\DateTime\Temporal;
+
+/**
+ * @unreleased
+ */
+class CampaignOverviewStatistics implements RestRoute
+{
+    /** @var string */
+    protected $endpoint = 'campaign-overview-statistics';
+
+    /**
+     * @unreleased
+     */
+    public function registerRoute()
+    {
+        register_rest_route(
+            'give-api/v2',
+            $this->endpoint,
+            [
+                [
+                    'methods' => 'GET',
+                    'callback' => [$this, 'handleRequest'],
+                    'permission_callback' => function () {
+                        return current_user_can('manage_options');
+                    },
+                ],
+                'args' => [
+                    'campaignId' => [
+                        'type' => 'integer',
+                        'required' => true,
+                        'validate_callback' => 'is_numeric',
+                    ],
+                    'rangeInDays' => [
+                        'type' => 'integer',
+                        'required' => false,
+                        'validate_callback' => 'is_numeric',
+                        'default' => 0, // Zero to mean "all time".
+                    ],
+                ],
+            ]
+        );
+    }
+
+    /**
+     * @unreleased
+     */
+    public function handleRequest($request)
```

Comment:
Let's a `@throws Exception` tag and change the return type of this function to `: WP_REST_Response` - we also need to adapt the body method to return it instead of an array.

Reply:
Resolved, see [290e920](https://github.com/impress-org/givewp/pull/7545/commits/290e9206d461b84001894a97673dbf1afbc1a42d) and [4b45c15](https://github.com/impress-org/givewp/pull/7545/commits/4b45c15909c3d10e81d395d37152bb2816ef35a5).